### PR TITLE
[NFC][Clang] Move set functions out BranchProtectionInfo.

### PR DESCRIPTION
**Pull request by DanielKristofKiss**
To reduce build times move them to TargetCodeGenInfo.
Refactor of #98329

### CODE REVIEW

File: clang/lib/CodeGen/Targets/AArch64.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ class AArch64TargetCodeGenInfo : public TargetCodeGenInfo {
       }
     }
     auto *Fn = cast<llvm::Function>(GV);
-    BPI.setFnAttributes(*Fn);
+    CGM.getTargetCodeGenInfo().setFnAttributes(BPI, *Fn);
```
Comment (nikic) [inline clang/lib/CodeGen/Targets/AArch64.cpp:136]:
```suggestion
    setFnAttributes(BPI, *Fn);
```
We're inside AArch64TargetCodeGenInfo here, so I think this is just a member function?

File: clang/lib/CodeGen/TargetInfo.h [hunk 2]
```diff
@@ -413,6 +414,12 @@ class TargetCodeGenInfo {
     return nullptr;
   }
 
+  void setFnAttributes(const TargetInfo::BranchProtectionInfo &BPI,
```
Comment (nikic) [inline clang/lib/CodeGen/TargetInfo.h:417]:
```suggestion
  void setBranchProtectionFnAttributes(const TargetInfo::BranchProtectionInfo &BPI,
```
Or so? The current name seems a bit too generic.

Comment (DanielKristofKiss) [reply clang/lib/CodeGen/TargetInfo.h]: agree, added.

File: clang/lib/CodeGen/Targets/AArch64.cpp (hunk 1 above)
Comment (DanielKristofKiss) [reply clang/lib/CodeGen/Targets/AArch64.cpp]: oh right.

File: clang/lib/CodeGen/Targets/ARM.cpp [hunk 3]
```diff
@@ -152,7 +152,7 @@ class ARMTargetCodeGenInfo : public TargetCodeGenInfo {
               diag::warn_target_unsupported_branch_protection_attribute)
               << Arch;
         } else {
-          BPI.setFnAttributes(*Fn);
+          setBranchProtectionFnAttributes(BPI, (*Fn));
```
Comment (efriedma-quic) [inline clang/lib/CodeGen/Targets/ARM.cpp:155]: Unnecessary parens.

**Review by efriedma-quic — APPROVED** [review]
LGTM with one minor comment

**Comment by DanielKristofKiss** [comment]
#83277 needs the `setBranchProtectionFnAttributes` function but that call is from a static function so let's make it static.

**Pull request merged**